src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -786,29 +787,34 @@ static function (ChildDefinition $definition, AsPeriodicTask|AsCronTask $attribu
         }
 
         $container->registerForAutoconfiguration(CompilerPassInterface::class)
-            ->addTag('container.excluded.compiler_pass')->addTag('container.excluded')->setAbstract(true);
+            ->addTag('container.excluded', ['source' => 'because it\'s a compiler pass'])->setAbstract(true);
+        $container->registerForAutoconfiguration(Constraint::class)
+            ->addTag('container.excluded', ['source' => 'because it\'s a validation constraint'])->setAbstract(true);
         $container->registerForAutoconfiguration(TestCase::class)
-            ->addTag('container.excluded.test_case')->addTag('container.excluded')->setAbstract(true);
+            ->addTag('container.excluded', ['source' => 'because it\'s a test case'])->setAbstract(true);
+        $container->registerForAutoconfiguration(\UnitEnum::class)
```

Comment:
Looking for edge cases, an enum value could be used as a service. But that's really weird.